s3/pkg/datastore/driver/factory.go

```diff
@@ -0,0 +1,23 @@
+package driver
+
+import (
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	exp "github.com/opensds/multi-cloud/s3/pkg/exception"
+)
+
+type DriverFactory interface {
+	CreateDriver(detail *backendpb.BackendDetail) (StorageDriver, error)
+}
+
+var driverFactoryMgr = make(map[string]DriverFactory)
+
+func RegisterDriverFactory(driverType string, factory DriverFactory) {
+	driverFactoryMgr[driverType] = factory
+}
+
+func CreateStorageDriver(driverType string, detail *backendpb.BackendDetail) (StorageDriver, error) {
+	if factory, ok := driverFactoryMgr[driverType]; ok {
+		return factory.CreateDriver(detail)
```

Comment:
imho, the factory.CreateDriver should NOT return the driver directly to the client...there should be some book keeping in the factory, then return the StorageDriver to the caller...This way, the factory can control various aspects of the StorageDriver like 
- is it a singleton or not ?
- lifecycle policies of the StorageDriver
- setup and tearDown of the StorageDriver

Reply:
Hi rhaskarpos, thanks for your suggestion. and StorageDriver is managed by the factory and the  factory controls whether the driver is singleton or not, currently, yig storage driver is a singleton, and each StorageDriver has an interface called Close, after the caller uses the driver, he should close it.